tools/ramdump/dumpParser.py

```diff
@@ -0,0 +1,700 @@
+
+		# Verify the Addresses passed
+		if debug:
+			print "fp:{0:x}  sp low:{1:x} high:{2:x}".format(frame.fp, frame.sp, high)
+
+		# Check current frame pointer is within bounds of Stack*/
+		if (fp < (low + 12) or fp + 4 >= high):
+			print "End of stack frame"
+			return -1
+
+		# If the Ramdump is given, Read contents of address from the RAMDUMP file
+		# by using read_word API
+		if self.log_file is None:
+			fp_is_at = self.read_word(frame.fp - 12)
+			sp_is_at = self.read_word(frame.fp - 8)
+			pc_is_at = self.read_word(frame.fp - 4)
+		else:
+			# If only log file is given, Read contents of addr from the populated stack_table
+			fp_is_at = self.stack_lookup(frame.fp - 12)
+			sp_is_at = self.stack_lookup(frame.fp - 8)
+			pc_is_at = self.stack_lookup(frame.fp - 4)
+
+		frame.fp = fp_is_at
+		frame.sp = sp_is_at
+		frame.pc = pc_is_at
+		return 0
+
+	# Function to walk the stack frame when framepointer is enabled
+	def find_stackframe_using_framepointer(self, frame, extra_str='', output_file=None):
+		while True:
+
+			offset = 0
+			# If PC is Null, prefetch abort or Something wrong, Break
+			if frame.pc is None:
+				break
+			if(frame.pc == 0):
+				break
+
+			# Read the Symbol name and offset for a specific PC address from symbol table
+			r = self.symbol_table_lookup(frame.pc)
+			if r is None:
+				symname = 'UNKNOWN'
+				offset = 0x0
+			else:
+				symname, offset = r # Update both symbol name and offset
+
+			pstring = (extra_str + '[<{0:x}>] {1}+0x{2:x}'.format(frame.pc, symname, offset))
+
+			if output_file:
+				out_file.write(pstring + '\n')
+			else:
+				print(pstring)
+
+			# Unwind the complete Stack with the help of Frame pointer register
+			ret = self.unwind_frame_using_framepointer(frame)
+			if ret < 0:
+				break
+
+	# Function to backtrace the stack in Generic method, which uses frame pointer register
+	# First function to get called to unwind
+	def unwind_backtrace_with_framepointer(self, fp, sp, lr, pc, ss):
+		frame = self.Stackframe(fp,sp,lr,pc)
+		frame.fp = fp
+		frame.sp = sp
+		frame.lr = lr
+		frame.pc = pc
+		self.stacksize = ss
+		self.find_stackframe_using_framepointer(frame)
+
+def usage():
+	print '*************************************************************'
+	print '\nUsage: %s [OPTIONS]\n' % sys.argv[0]
+	print '\t-r, --ramdump_file              ramdump_file along with path'
+	print '\t-e, --elf                       tinyara elf file along with path'
+	print '\t-G, --gdb_path                  Enter gdb tool path'
+	print '\t-N, --nm_path                   Enter nm tool path'
+	print '\t-E, --readelf_path              Enter readelf tool path'
+	print '\t-t, --log_file                  Enter Logfile path'
+	print '\t-h, --help                      Show help'
+	print ''
+	print 'syntax :'
+	print '--------'
+	print 'python dumpParser.py -r Filename_ramBaseAddr_ramEndAddr.bin  -e <Tinyara elf path> -G <Gdb path> -N < NM path> '
```

Comment:
At above, you use ```% sys.argv[0]```. But not in here. Above is better I think.

Reply:
Ok. I modified it